include/ck_tile/host/kernel_launch.hpp

```diff
@@ -28,18 +28,47 @@ __launch_bounds__(Kernel::kBlockSize, MinBlockPerCu)
 #endif
 }
 
+template <typename Arch, int MinBlockPerCu, typename Kernel, typename... Args>
+#if CK_TILE_USE_LAUNCH_BOUNDS
+__launch_bounds__(Kernel::kBlockSize, MinBlockPerCu)
+#endif
+    __global__ void kentry(Args... args)
+{
+#if defined(__HIP_DEVICE_COMPILE__)
+    Kernel{}(args...);
+#else
+    (..., (ignore = args, 0));
+#endif
+}
+
 //
 // return a anonymous functor(lambda) to be called later
 // the KernelImpl should be a class without non-static data member, or let's say
 // can be instantiate with "KernelImpl{}"
 //
 // the "static __device__ operator()(some_arg)" is the entry point of KernelImpl
 //
-template <int MinBlockPerCu = CK_TILE_MIN_BLOCK_PER_CU, typename KernelImpl, typename... Args>
+// Arch can be used to support linking multiple object files that have the same kernel compiled for
```

Comment:
what will happen if we don't add arch in kenerl template? link error or something else?
I think compiler should have handled kernel on multi device.

Reply:
Let's assume that we generate the same kernel (same tile size and pipeline) for different architectures, gfx9 and gfx12. Each is built as a separate file: `fmha..._gfx9.o` and `fmha..._gfx12.o`. So these two files use the same symbol for the kernel built for different architectures, the linker does not combine them into one "fat" symbol but keeps only one of them in the resulting binary.

In this case we can get a **runtime** error that there is not kernel for the current device (and on another device it works).